sdk/eventhubs/azure-messaging-eventhubs/inc/azure/messaging/eventhubs/producer_client.hpp

```diff
@@ -92,13 +92,27 @@ namespace Azure { namespace Messaging { namespace EventHubs {
         std::shared_ptr<Azure::Core::Credentials::TokenCredential> credential,
         ProducerClientOptions options = {});
 
-    ~ProducerClient()
+    ~ProducerClient() {}
+
+    /** @brief Close all the connections and sessions.
+     *
+     * @param context Context for the operation can be used for request cancellation.
+     */
+    void Close(Azure::Core::Context const& context = {})
     {
       for (auto& sender : m_senders)
       {
-        sender.second.Close();
+        sender.second.Close(context);
       }
       m_senders.clear();
+      //    for (auto& session : m_sessions)
+      //    {
+      // session.second.Close(context);
+      // }
+      //    for (auto& connection : m_connections)
+      //    {
+      // connection.second.Close(context);
+      // }
```

Comment:
Is there an expecation this commented code will come back?  Add a note if so, otherwise please remove.

Reply:
It might, once I figure out a story for what "Close" means. But as it currently stands, the commented out code is going to cause massive pain and suffering.

But I don't want to lose it because it's important.